src/blueapi/worker/task_worker.py

```diff
@@ -138,19 +138,18 @@ def get_task_by_id(self, task_id: str) -> TrackableTask | None:
         return self._tasks.get(task_id)
 
     def get_tasks_by_status(self, status: TaskStatusEnum) -> list[TrackableTask]:
-        if status == TaskStatusEnum.RUNNING:
-            return [
-                task
-                for task in self._tasks.values()
-                if not task.is_pending and not task.is_complete
-            ]
-        elif status == TaskStatusEnum.PENDING:
-            return [task for task in self._tasks.values() if task.is_pending]
-        elif status == TaskStatusEnum.COMPLETE:
-            return [task for task in self._tasks.values() if task.is_complete]
-        return []
-
-    def get_active_task(self) -> TrackableTask[Task] | None:
+        # Define filters for each status
+        status_filter = {
+            TaskStatusEnum.RUNNING: lambda task: not task.is_pending
+            and not task.is_complete,  # noqa: E501
+            TaskStatusEnum.PENDING: lambda task: task.is_pending,
+            TaskStatusEnum.COMPLETE: lambda task: task.is_complete,
+        }.get(status, lambda task: False)  # Default to a filter that returns no tasks
+
+        # Apply the filter to _tasks values
+        return [task for task in self._tasks.values() if status_filter(task)]
```

Comment:
Should: I don't think this any clearer, please revert? Alternatively see if it's any better with `match` statements